src/main/java/org/sonar/plugins/stash/issue/MarkdownPrinter.java

```diff
@@ -63,15 +83,24 @@ public static String printReportMarkdown(PullRequestRef pr, String stashURL, Str
     String stashRepo     = pr.repository();
     int pullRequestId = pr.pullRequestId();
 
-    if ((report.getIssues() == null) || (report.getIssues().isEmpty() && coverageReport.getLoweredIssues().isEmpty())) {
+    List<Issue> coverageIssues = new ArrayList<>();

```

Comment:
I mean, you build the list of different issues from report.getIssues(), but you can create 2 methods in Report (new Class?): getCoverageIssues() and getGeneralIssues() and move the code below to it.

I think it a good idea to claim that Report class is responsible to return a list of issues it holds filtered for our need (by type, type severity etc.)

I remember I saw this... in master?